File: examples/hello.js
// Hello World — Node.js on MerlionOS.
// Run: run-user node hello.js

console.log("Hello from Node.js on MerlionOS!");
console.log(`Node.js ${process.version}`);
console.log(`PID: ${process.pid}`);

// Promises + async
async function main() {
    // Array methods
    const nums = [1, 2, 3, 4, 5];
    const squares = nums.map(n => n * n);
    console.log(`Squares: ${squares}`);

    // Object destructuring
    const info = { os: "MerlionOS", lang: "JavaScript", year: 2026 };
    const { os, lang } = info;
    console.log(`Running ${lang} on ${os}`);

    // Timer
    const start = Date.now();
    await new Promise(resolve => setTimeout(resolve, 100));
    console.log(`Slept ${Date.now() - start}ms`);

    console.log("All Node.js tests passed!");
}

main().catch(console.error);

File: examples/hello.py
#!/usr/bin/env python3
"""Hello World — Python on MerlionOS."""

import sys
import os
import time

print("Hello from Python on MerlionOS!")
print(f"Python {sys.version}")
print(f"Platform: {sys.platform}")
print(f"PID: {os.getpid()}")
print(f"CWD: {os.getcwd()}")

# File I/O
with open("/tmp/python_test.txt", "w") as f:
    f.write("Python file I/O works on MerlionOS!\n")

with open("/tmp/python_test.txt") as f:
    print(f"File: {f.read().strip()}")

# Data structures
data = {"os": "MerlionOS", "lang": "Python", "year": 2026}
print(f"Dict: {data}")

nums = [x**2 for x in range(10)]
print(f"Squares: {nums}")

# Timing
start = time.monotonic()
time.sleep(0.1)
elapsed = time.monotonic() - start
print(f"Slept {elapsed:.3f}s")

print("All Python tests passed!")
